mlrun/api/db/sqldb/db.py

```diff
@@ -1354,38 +1411,8 @@ def _add_utc_timezone(obj, attribute_name):
     def _transform_feature_set_model_to_schema(
         feature_set_record: FeatureSet, tag=None,
     ) -> schemas.FeatureSet:
-
-        feature_set = schemas.FeatureSetRecord.from_orm(feature_set_record)
-
-        feature_set_metadata = schemas.FeatureSetMetadata(
-            name=feature_set.name,
-            updated=feature_set.updated,
-            labels=transform_label_list_to_dict(feature_set.labels),
-            uid=feature_set.uid,
-        )
-        feature_set_spec = schemas.FeatureSetSpec(entities=[], features=[])
-        for feature in feature_set.features:
-            feature_set_spec.features.append(
-                schemas.Feature(
-                    name=feature.name,
-                    value_type=feature.value_type,
-                    labels=transform_label_list_to_dict(feature.labels),
-                )
-            )
-        for entity in feature_set.entities:
-            feature_set_spec.entities.append(
-                schemas.Entity(
-                    name=entity.name,
-                    value_type=entity.value_type,
-                    labels=transform_label_list_to_dict(entity.labels),
-                )
-            )
-
-        feature_set_resp = schemas.FeatureSet(
-            metadata=feature_set_metadata,
-            spec=feature_set_spec,
-            status=feature_set.status,
-        )
+        feature_set_full_dict = feature_set_record.full_object
+        feature_set_resp = schemas.FeatureSet(**feature_set_full_dict)
```

Comment:
How come all the labels transformation is not needed anymore ?

Reply:
That's because the response is now generated from the `.full_object` property, which is the full object stored as a dictionary. This means that the labels do not need to be transformed from the DB table form to dictionary form anymore. This is required since we want to maintain any additional data that the user provided as input (`extra` fields), and return them back. I saw no point in doing the rebuild from the full object and then re-applying information from the DB table fields, since they must be in-sync. If there's a mismatch then it means there's a bug in the create/update logic and it needs to be fixed there.